invenio_migrator/legacy/users.py

```diff
@@ -21,39 +21,71 @@
 # In applying this license, CERN does not
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
-
 """Community dump functions."""
 
 from __future__ import absolute_import, print_function
 
+from collections import namedtuple
 from .utils import dt2iso_or_empty
 
 
+def _get_users_invenio12(*args, **kwargs):
+    """Get user accounts Invenio 1."""
+    from invenio.dbquery import run_sql, deserialize_via_marshal
+    User = namedtuple('User', [
+        'id', 'email', 'password', 'password_salt', 'note', 'full_name',
+        'settings', 'nickname', 'last_login'
+    ])
+    users = run_sql(
+        'SELECT id, email, password, note, settings, nickname, last_login'
+        ' FROM user',
+        run_on_slave=True)
+    return len(users), [
+        User(
+            id=user[0],
+            email=user[1],
+            password=user[2].decode('latin1'),
```

Comment:
Hmm, `latin1` may not be the safest general default here, but it should match all the targeted users (or installlations) of `invenio-migrator` I think?

Reply:
Actually, I have just tried this PR and found some problems. Since in Invenio 1 the column is blob and in Invenio 3 it is a string, when dumping and loading the password, in some cases I get an error with a password like:

```
ipdb> obj.password
u'$invenio-aes$yursh@ihep.ac.cn$\xed?\x95x\xc7;\xc9\x15p\x17b\x94\xbfA\xd2\xf1\xccssIM\xa6\x00kxm\x85a|mW\xf6'
```
```
ValueError: A string literal cannot contain NUL (0x00) characters.
```

Since Postgresql does not allow the NUL character in text fields.

Reply:
And for those passwords that do not give an error, I don't think they are being hashed properly in Invenio 3. After migrating my user from Invenio 1 to Invenio 3, using my same password does not work.

Reply:
Honestly, I didn't test this part that much because we don't have any Invenio accounts on CDS, for sure something that I need to look into.